.github/workflows/release.yml

```diff
@@ -8,23 +8,16 @@ jobs:
   goreleaser:
     runs-on: ubuntu-latest
     steps:
-      -
-        name: Checkout
-        uses: actions/checkout@v2
-      -
-        name: Unshallow
-        run: git fetch --prune --unshallow
-      -
-        name: Set up Go
-        uses: actions/setup-go@v1
+      - name: Checkout
+        uses: actions/checkout@v4
```

Comment:
maybe add fetch-depth: 0 if you are unsure about tags being present. In the default case (when you release on tags, there will be a tag associated to the ref being checked out by git. You can test this by creating a dummy-tag on your latest commit in this PR.